### PR TITLE
DOC: Clarify `addfont` (globalness & persistence)

### PR DESCRIPTION
**Pull request by OverLordGoldDragon**
## PR Summary

Fixes https://discourse.matplotlib.org/t/how-to-add-fonts/23221

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -1114,7 +1114,10 @@ def __init__(self, size=None, weight='normal'):
     def addfont(self, path):
         """
         Cache the properties of the font at *path* to make it available to the
-        `FontManager`.  The type of font is inferred from the path suffix.
+        `FontManager`. The type of font is inferred from the path suffix.
+
+        To cache globally and persistently, instead use
+        `matplotlib.font_manager.fontManager.addpath(path)`.
```
Comment (oscargus) [inline lib/matplotlib/font_manager.py:1121]: Either use double backtick to create fixed-width text or remove the argument to get a link to the merhod. Putting a tilde first will abbreviate a the text to just the method.

File: lib/matplotlib/font_manager.py [hunk 2]
```diff
@@ -1114,7 +1114,10 @@ def __init__(self, size=None, weight='normal'):
     def addfont(self, path):
         """
         Cache the properties of the font at *path* to make it available to the
-        `FontManager`.  The type of font is inferred from the path suffix.
+        `FontManager`. The type of font is inferred from the path suffix.
+
+        To cache globally and persistently, instead use
+        ``matplotlib.font_manager.fontManager.addpath(path)``.
```
Comment (timhoffm) [inline lib/matplotlib/font_manager.py:1120]:
```suggestion
        The cache only exists for the current session. To cache globally and
        persistently, instead use 
        ``matplotlib.font_manager.fontManager.addpath(path)``.
```

**Comment by tacaswell** [comment]
I am not completely sure that we want to document this.  There are some conditions under which we will automatically re-generate the font cache and "delete the font cache, restart Python" is a common suggestion for fixing font issues so this is only mostly persistent.

**Comment by OverLordGoldDragon** [comment]
My use case is ReadTheDocs, where I don't know how to "restart Python" (and the need to do this would cause trouble for others using my package on remote builds), so I need something to make it work from the getgo.

>  There are some conditions under which we will automatically re-generate the font cache

Meaning undoing `fontManager.addpath()`? So the only permanent solution is to install a font to the system? That's even more complicated, so instead the docstring could mention this caveat - my approach is to check `font_manager.get_font_names()` upon `import my_pkg` and `addfont()` if the font's not there.

**Comment by tacaswell** [comment]
@OverLordGoldDragon Can you explain a bit more what you use case is?  It sounds like you are using a Python package to install fonts which while it "works" (and is something we do!) I have concerns about mpl supporting fonts installed by third party packages writing back into the font cache.

To make sure we are all on the same page:

 - finding fonts is slow (so slow we have a timer to warn users it can be slow)
 - when we find (system) fonts we write a json file to a cache directory in the mplconfig directories
 - this file is named such that it tracks an internal version number (which is independent of the library version) if we change the format of that file.
 - this file is shared across all Python processes run on that machine under that user (well, there is a `MPLCONFIGDIR` env so really processes that share the configuration directory).
 - adding paths that maybe in a (transient) environment to this cache seems like it will lead to other problems down the line (as the env gets deleted and now we have a broken font).

Reading the source, I am also not sure that the added text is correct.  We only cache to the disk via https://github.com/matplotlib/matplotlib/blob/68c78c9fb1deeb5ddc4951efbe32445e455bb4a7/lib/matplotlib/font_manager.py#L1619-L1637 and 
https://github.com/matplotlib/matplotlib/blob/68c78c9fb1deeb5ddc4951efbe32445e455bb4a7/lib/matplotlib/font_manager.py#L1029-L1050 which I do not think calling `addpath` will trigger.

-----

We do search user space folders on all three platforms (see https://github.com/matplotlib/matplotlib/blob/68c78c9fb1deeb5ddc4951efbe32445e455bb4a7/lib/matplotlib/font_manager.py#L130-L167 ) so it may be better to drop your font there if you want it to "just work" (but this is pushing what is done by a Python package to an extreme).

**Comment by tacaswell** [comment]
I've moved this to draft because as it stands this refers to a non-existent method and I think the discussion about persistence is incorrect.

https://matplotlib.org/stable/api/font_manager_api.html does explain there is a global (module level) instance of `FontManger`, maybe that needs to be clearer?  If we do want a comment about how to register your own ad-hoc fonts, I think it should go at the module level not on the method like proposed here.

**Comment by OverLordGoldDragon** [comment]
Thanks @story645 It'll be a while before I get back to recalling what this Issue was about but I don't intend to work on this PR regardless the outcome.

**Pull request closed without merging**